hphp/runtime/ext/std/ext_std_file.cpp

```diff
@@ -588,9 +588,21 @@ Variant HHVM_FUNCTION(file_put_contents,
                       int flags /* = 0 */,
                       const Variant& context /* = null */) {
   CHECK_PATH(filename, 1);
+
+  char mode[3] = "wb";
+  if (flags & PHP_FILE_APPEND) {
+    mode[0] = 'a';
+  } else if (flags & LOCK_EX) {
+    // Open in "create" mode (writing only, create if needed, no truncate)
+    // so that the file is not modified before we attempt to aquire the
+    // requested lock.
+    mode[0] = 'c';
```

Comment:
I don't think 'c' does what you think it does...

```
NOTES
   Glibc Notes
       The GNU C library allows the following extensions for the string speci-
       fied in mode:

       c (since glibc 2.3.3)
              Do not make the open operation, or  subsequent  read  and  write
              operations, thread cancellation points.
```